src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/FixAllSpanMappingService/AbstractFixAllSpanMappingService.cs

```diff
@@ -0,0 +1,96 @@
+                return await GetFixAllSpansIfWithinGlobalStatementAsync(document, diagnosticSpan, fixAllScope, cancellationToken).ConfigureAwait(false);
+
+            if (fixAllScope == FixAllScope.ContainingMember)
+            {
+                return ImmutableDictionary<Document, ImmutableArray<TextSpan>>.Empty
+                    .Add(document, ImmutableArray.Create(decl.FullSpan));
+            }
+            else
+            {
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var symbol = semanticModel.GetDeclaredSymbol(decl, cancellationToken);
+                if (symbol?.DeclaringSyntaxReferences.Length > 1)
+                {
+                    var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+                    var builder = PooledDictionary<Document, ArrayBuilder<TextSpan>>.GetInstance();
+                    foreach (var syntaxRef in symbol.DeclaringSyntaxReferences)
+                    {
+                        var documentForLocation = document.Project.GetDocument(syntaxRef.SyntaxTree);
+                        Contract.ThrowIfNull(documentForLocation);
+                        var root = await syntaxRef.SyntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+                        var partialDeclSpan = syntaxFacts.GetContainingTypeDeclaration(root, syntaxRef.Span.Start)!.FullSpan;
+                        builder.MultiAdd(documentForLocation, partialDeclSpan);
+                    }
+
+                    return builder.ToImmutableMultiDictionaryAndFree();
+                }
+                else
+                {
+                    return ImmutableDictionary<Document, ImmutableArray<TextSpan>>.Empty
+                        .SetItem(document, ImmutableArray.Create(decl.FullSpan));
+                }
+            }
+        }
+
+        private static async Task<SyntaxNode?> GetContainingMemberOrTypeDeclarationAsync(
+            Document document,
+            FixAllScope fixAllScope,
+            TextSpan span,
+            CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfFalse(fixAllScope is FixAllScope.ContainingMember or FixAllScope.ContainingType);
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+            var startContainer = fixAllScope == FixAllScope.ContainingMember
+                ? syntaxFacts.GetContainingMemberDeclaration(root, span.Start)
+                : syntaxFacts.GetContainingTypeDeclaration(root, span.Start);
+
+            if (startContainer == null)
+                return null;
+
+            if (fixAllScope == FixAllScope.ContainingMember && !syntaxFacts.IsMethodLevelMember(startContainer))
+                return null;
+
+            if (span.IsEmpty)
```

Comment:
when is span empty?

Reply:
Diagnostic with zero span size and non-zero start?